### PR TITLE
feat: making group by compatible with url

### PR DESCRIPTION
**Pull request by anandtiwary**
## Description
Please include a summary of the change, motivation and context.
feat: making group by compatible with url

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

**Review by palbizu — APPROVED** [review]
Looks good

File: projects/components/src/filtering/filter/filter-url.service.ts [hunk 1]
```diff
@@ -54,6 +55,19 @@ export class FilterUrlService {
     });
   }
 
+  public setUrlGroupBy(groupBy?: string[]): void {
```
[the remark this answers is not in the document]
Comment (anandtiwary) [reply projects/components/src/filtering/filter/filter-url.service.ts:58]: Need to change the name of this service. This should be more like filterBarUrlService. I have a follow up with related change

[the remark this answers is not in the document]
Comment (aaron-steinfeld) [reply projects/components/src/filtering/filter/filter-url.service.ts:58]: But explorer manages its own group by, while its filters are managed by a filter bar component which uses this, I believe. So it seems like we're setting up multiple ways of doing it? Should we put explorers group by into here then?

[the remark this answers is not in the document]
Comment (anandtiwary) [reply projects/components/src/filtering/filter/filter-url.service.ts:58]: going to merge this now unless you have an objection.

**Pull request merged**